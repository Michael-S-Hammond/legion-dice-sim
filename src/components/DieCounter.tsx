import '../css/DieCounter.css';

import React from 'react';

type DieCounterProps = {
    count: number,
    styleName: string,
    onClick: () => void,
};

class DieCounter extends React.Component<DieCounterProps> {
    constructor(props : DieCounterProps) {
        super(props);
    }

    render() {
        return (
            <button
                className={`btn border border-secondary rounded-lg die-counter mx-1 ${ this.props.styleName }`}
                onClick={() => this.props.onClick()}>
                    { this.props.count }
            </button>
        );
    }
}

export default DieCounter;
