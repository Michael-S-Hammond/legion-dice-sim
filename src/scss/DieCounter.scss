.die-counter {
    display: flex;
    width: 50px;
    height: 50px;
    justify-content: center;
    align-items: center;
    font-weight: bold;
    font-size: larger;
}
