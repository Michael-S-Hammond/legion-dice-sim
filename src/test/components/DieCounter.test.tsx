import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import DieCounter from '../../components/DieCounter';

describe('DieCounter', () => {
    it('matches the snapshot', () => {
        function onClick() {
        }

        const counter = <DieCounter
                count={4}
                styleName='btn-dark'
                onClick={onClick}
            ></DieCounter>
        const snapshot = create(counter);
        expect(snapshot.toJSON()).toMatchSnapshot();
    });

    it('handles being clicked', () => {
        const onButtonClickMock = jest.fn();

        let wrapper = shallow(<DieCounter
            count={4}
            styleName='btn-dark'
            onClick={onButtonClickMock}
        ></DieCounter>);
        wrapper.find('.btn').simulate('click');

        expect(onButtonClickMock).toHaveBeenCalledTimes(1);
    })
});
