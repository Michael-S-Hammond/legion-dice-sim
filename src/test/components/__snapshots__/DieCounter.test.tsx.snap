// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DieCounter matches the snapshot 1`] = `
<button
  className="btn border border-secondary rounded-lg die-counter mx-1 btn-dark"
  onClick={[Function]}
>
  4
</button>
`;
